src/main/java/org/elasql/perf/tpart/metric/MetricCollector.java

```diff
@@ -9,11 +11,15 @@
  * 
  * @author Yu-Shan Lin
  */
+@SuppressWarnings("restriction")
```

Comment:
what is this for?

Reply:
The original warning is `Access restriction: The method 'OperatingSystemMXBean.getSystemCpuLoad()' is not API (restriction on required library 'C:\Program Files\Java\jdk1.8.0_241\jre\lib\rt.jar')`

Based on what I found in stackoverflow, it is just Eclipse trying to protect us from using classes that are not part of the official JDK API.

Here are two package `com.sun.management.OperatingSystemMXBean` & `java.lang.management.OperatingSystemMXBean`. Both supports to get some system info.

The package from `java.lang` doesn't provide APIs such as `getSystemCPULoad` and `getProcessCpuLoad`. It just simply provides `getSystemLoadAverage`. However, the package from `com.sum` provides all we need.

Reply:
Got it.